### PR TITLE
feat: integrate lit sdk for decrypt / encrypt

### LINKED ISSUE #220: Integrate LIT SDK
**User Story:**
As a RM engineer
I want to integrate the LIT protocol SDK with Round Manager
So that I can generate public/private key pairs for PII encryption

**Acceptance Criteria**
- Integrate [LIT SDK](https://developer.litprotocol.com/)
- Configure LIT for the storage of public/private key pairs (includes configuring so that only round operators can decrypt for their round). The RM will generate the keyPair using https://developer.mozilla.org/en-US/docs/Web/API/Crypto/subtle and then save the keyPair generated to Lit using an access control config (ex: address). NOTE: this should allow anyone access to the public key, but only round operators have access to private key.

**Product & Design Links:**

**Tech Details:**
The grant hub team has some examples / tests available here:
- https://gist.github.com/gravityblast/141d294e22a99459c202437f854b9009
- https://github.com/codenamejason/cryptography-lit-example

**Open Questions:**

**Notes/Assumptions:**
- This includes writing test coverage for the functionality added in the ticket, if necessary

### PR DESCRIPTION
**Pull request by thelostone-mc**
##### Description

- creates a working implementation of encrypt / decrypting using lit
- added documenation on how we would encrypt / decrypt data
- locally tested

##### Refers/Fixes

closes #220 

##### Testing

Deploy

### CODE REVIEW

File: packages/round-manager/src/features/api/lit.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    // Encrypting Content and generating symmetric key
+    const { encryptedString, symmetricKey } = await LitJsSdk.encryptString(content);
+    console.log("Encrypted the key pair");
+    console.log("Encrypted String", encryptedString);
+    console.log("Symmetric Key", symmetricKey);
+
+    // Saving the Encrypted Content to the Lit Nodes
+    const encryptedSymmetricKey = await this.litNodeClient.saveEncryptionKey({
+      accessControlConditions: this.isRoundOperatorAccessControl(),
```
Comment (thelostone-mc) [inline packages/round-manager/src/features/api/lit.ts:99]: Shouldn't this evmContractConditions as well ?

File: packages/round-manager/docs/EncryptingPII.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+- `encryptedSymmetricKey`
+
+Here is an example of the entire flow in action.
+
+```javascript
+
+// Init LIT config
+const lit = new Lit({
+  chain: "goerli",
+  contract: "0x22c0e3EDc90f6A890A259130B416Cd5F3Ee4Aca0"
```
Comment (shavinac) [inline packages/round-manager/docs/EncryptingPII.md:37]: just to make sure i have this right - this address is a round contract, and LIT only decrypts for addresses who are round operators for that particular round?

**Review by shavinac — APPROVED** [review]

File: packages/round-manager/src/features/api/lit.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    // Obtain Auth Signature to verify signer is wallet owner
+    const chain = this.chain;
+    const authSig = await LitJsSdk.checkAndSignAuthMessage({ chain });
+
+    // Encrypting Content and generating symmetric key
+    const { encryptedString, symmetricKey } = await LitJsSdk.encryptString(content);
+    // console.log("Encrypted the key pair");
+    // console.log("Encrypted String", encryptedString);
+    // console.log("Symmetric Key", symmetricKey);
+
```
Comment (chibie) [inline packages/round-manager/src/features/api/lit.ts:94]: can we remove console log comments like these?

File: packages/round-manager/src/features/api/lit.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
+import { Buffer } from 'buffer';
+const LitJsSdk = require('lit-js-sdk');
```
Comment (chibie) [inline packages/round-manager/src/features/api/lit.ts:2]: why `require` and not `import`?

**Review by chibie — CHANGES_REQUESTED** [review]
few comments + merge conflicts that need to be resolved

File: packages/round-manager/src/features/api/lit.ts (hunk 4 above)
Comment (chibie) [reply packages/round-manager/src/features/api/lit.ts:2]: i know why now..

**Review by chibie — APPROVED** [review]

**Pull request merged**